src/libraries/System.ComponentModel.Composition/src/System/ComponentModel/Composition/Hosting/AssemblyCatalog.cs

```diff
@@ -543,6 +543,8 @@ private void ThrowIfDisposed()
         private string GetDisplayName() =>
             $"{GetType().Name} (Assembly=\"{Assembly.FullName}\")";   // NOLOC
 
+        [UnconditionalSuppressMessage("Single file", "IL3000: Avoid accessing Assembly file path when publishing as a single file",
+            Justification = "Setting a CodeBase is not single file dangerous")]
```

Comment:
If setting a CodeBase isn't a problem - then why is it annotated as such?

Reply:
The analyzer treats the whole property as dangerous because getting information from it will be empty in single-file

Reply:
RequiresUnreferencedCode can be applied to just one of the accessors - doesn't it work the same for RequiresAssemblyFiles as well?

Reply:
The analyzer recognizes a special range of dangerous patterns in which RequiresAssemblyFiles is irrelevant since they will generate IL3000 and not IL3002 (even if annotated with RequiresAssemblyFiles). I'm looking into how to make it warn only on the getter.

Reply:
RequiresUnreferencedCode can't be applied to a property. It is only valid on a method. So you need to annotate 1 or both accessors if you want it on a property.

Reply:
Actually, I got rid of CodeBase/EscapedCodeBase in the analyzer since they are [obsolete patterns](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Runtime/ref/System.Runtime.cs)
https://github.com/mono/linker/pull/1966

Reply:
There was a discussion in mono/linker#1966 the plan is:
Suppress the message and annotate CodeBase getter with RequiresAssemblyFiles
After the annotation flows, we delete the code in the analyzer that special cases CodeBase and EscapedCodeBase along with tests
After deletion of the code in the analyzer flows to runtime, we can delete the suppression in the runtime